src/hdf5_time_series_storage.jl

```diff
@@ -85,140 +85,414 @@ end
     storage::Hdf5TimeSeriesStorage,
     component_uuid::UUIDs.UUID,
-    label::AbstractString,
+    name::AbstractString,
     ts::TimeSeriesData,
-    columns = nothing,
 )
     check_read_only(storage)
     uuid = string(get_uuid(ts))
-    component_label = make_component_label(component_uuid, label)
+    component_name = make_component_name(component_uuid, name)
 
     HDF5.h5open(storage.file_path, "r+") do file
         root = _get_root(storage, file)
         if !HDF5.exists(root, uuid)
             HDF5.g_create(root, uuid)
             path = root[uuid]
-            @debug "Create new time series entry." uuid component_uuid label
-            path["data"] = TimeSeries.values(ts.data)
-            timestamps = [Dates.datetime2epochms(x) for x in TimeSeries.timestamp(ts.data)]
-            path["timestamps"] = timestamps
-            # Storing the UUID as an integer would take less space, but HDF5 library says
-            # arrays of 128-bit integers aren't supported.
-            path["components"] = [component_label]
-            if !isnothing(columns)
-                HDF5.attrs(path)["columns"] = string.(columns)
-            end
+            data = get_array_for_hdf(ts)
+            path["data"] = data
+            _write_time_series_attributes!(storage, ts, path)
+            path["components"] = [component_name]
+            @debug "Create new time series entry." uuid component_uuid name
         else
             path = root[uuid]
-            if !isnothing(columns)
-                if !HDF5.exists(HDF5.attrs(path), "columns")
-                    throw(ArgumentError("columns are specified but existing array does not have columns"))
-                end
-                existing = Symbol.(HDF5.read(HDF5.attrs(path)["columns"]))
-                if columns != existing
-                    throw(ArgumentError("columns do not match $columns $existing"))
-                end
-            end
-            @debug "Add reference to existing time series entry." uuid component_uuid label
-            _append_item!(path, "components", component_label)
+            @debug "Add reference to existing time series entry." uuid component_uuid name
+            _append_item!(path, "components", component_name)
         end
     end
+
+    return
+end
+
+"""
+Return a String for the data type of the forecast data, this implementation avoids the use of `eval` on arbitrary code stored in HDF dataset.
+"""
+function get_data_type(ts::TimeSeriesData)
+    data_type = eltype_data(ts)
+    if data_type <: CONSTANT
+        return "CONSTANT"
+    elseif data_type == POLYNOMIAL
+        return "POLYNOMIAL"
+    elseif data_type == PWL
+        return "PWL"
+    else
+        error("$data_type) is not supported in forecast data")
+    end
+end
+
+function eltype_data(ts::SingleTimeSeries)
+    return eltype(TimeSeries.values(ts.data))
+end
+
+function eltype_data(ts::TimeSeriesData)
+    return eltype(first(values(ts.data)))
+end
```

Comment:
This is broken for DeterministicSingleTimeSeries.  We should break it out by subtype.

Reply:
> 
> 
> @daniel-thom can we consider this ready for review?

I'll make a PR in the early afternoon with some minor fixes identified here.

I think we should do a review of the code coverage report and ensure that no critical logic is uncovered. Do you have time for that?

Reply:
resolved in #151